docs/content/guides/columns/column-filter.md

```diff
@@ -155,765 +450,2860 @@ import 'handsontable/dist/handsontable.full.min.css';
+      filtersRowsMap.setValueAtIndex(filtersRowsMap.indexedValues.length - 1, false);
+    });
   };
 
-  const toggleSelectAll = (event) => {
-    if (event.target.checked) {
-      setSelectedRowEntries(rowEntries);
+  return (
+    <HotTable
+      ref={hotTableComponentRef}
+      data={[
+        {
+          brand: 'Jetpulse',
+          model: 'Racing Socks',
+          price: 30,
+          sellDate: 'Oct 11, 2023',
+          sellTime: '01:23 AM',
+          inStock: false,
+        },
+        {
+          brand: 'Gigabox',
+          model: 'HL Mountain Frame',
+          price: 1890.9,
+          sellDate: 'May 3, 2023',
+          sellTime: '11:27 AM',
+          inStock: false,
+        },
+        {
+          brand: 'Camido',
+          model: 'Cycling Cap',
+          price: 130.1,
+          sellDate: 'Mar 27, 2023',
+          sellTime: '03:17 AM',
+          inStock: true,
+        },
+        {
+          brand: 'Chatterpoint',
+          model: 'Road Tire Tube',
+          price: 59,
+          sellDate: 'Aug 28, 2023',
+          sellTime: '08:01 AM',
+          inStock: true,
+        },
+        {
+          brand: 'Eidel',
+          model: 'HL Road Tire',
+          price: 279.99,
+          sellDate: 'Dec 2, 2023',
+          sellTime: '01:23 AM',
+          inStock: true,
+        },
+      ]}
+      columns={[
+        {
+          title: 'Brand',
+          type: 'text',
+          data: 'brand',
+        },
+        {
+          title: 'Model',
+          type: 'text',
+          data: 'model',
+        },
+        {
+          title: 'Price',
+          type: 'numeric',
+          data: 'price',
+          numericFormat: {
+            pattern: '$ 0,0.00',
+            culture: 'en-US',
+          },
+          className: 'htLeft',
+        },
+        {
+          title: 'Date',
+          type: 'date',
+          data: 'sellDate',
+          dateFormat: 'MMM D, YYYY',
+          correctFormat: true,
+          className: 'htRight',
+        },
+        {
+          title: 'Time',
+          type: 'time',
+          data: 'sellTime',
+          timeFormat: 'hh:mm A',
+          correctFormat: true,
+          className: 'htRight',
+        },
+        {
+          title: 'In stock',
+          type: 'checkbox',
+          data: 'inStock',
+          className: 'htCenter',
+        },
+      ]}
+      // enable the column menu
+      dropdownMenu={true}
+      // enable filtering
+      filters={true}
+      // `afterFilter()` is a Handsontable hook: it's fired after each filtering
+      afterFilter={exclude}
+      height="auto"
+      licenseKey="non-commercial-and-evaluation"
+    />
+  );
+};
+
+/* start:skip-in-preview */
+ReactDOM.render(<App />, document.getElementById('exampleExcludeRows'));
+/* end:skip-in-preview */
+```
 
-    } else {
-      setSelectedRowEntries([]);
-    }
+:::
+
+:::
+
+## Use filtering hooks
+
+You can run your code before or after filtering, using the following
```

Comment:
Again, what's the use case, where's the pain and benefit of using Handsontable hooks in this context? The part about server-side filtering, without an actual demo, is a bit blurry to me. Our technical support could provide some insights of how the before/after hooks are used in real-life cases.

Reply:
@AMBudnik are there any other uses of the `beforeFilter` and `afterFilter` hooks that come up in support cases (apart from blocking filtering on the client side by using `beforeFilter`)?

Reply:
There was one case when the developer needed to show a box if there's no data after filtering - https://jsfiddle.net/AMBudnik/91m4kLpr/.

![Zrzut ekranu 2023-04-19 o 07 56 24](https://user-images.githubusercontent.com/10757813/232979834-4c68ba4b-fe12-4d97-b824-9bf4d07c2386.png)

But mainly, it is about the logic 
1. Block the front-end filtering and proceed with filtering on back-end (but that one you've already discussed)
2. Retrieving the filtering states - http://jsfiddle.net/AMBudnik/1ebkcdan/ we use the `beforeFilter` hook to get the filtering conditions so the developer can reattach them when their user logs in. 
3. Collecting indexes of visible rows after filtering - https://jsfiddle.net/p9hz1wg3/1/ (but here I recommend asking devs which option to choose: `renderablePhysicalIndexesCache` or `notTrimmedIndexesCache`).

Reply:
Thanks @AMBudnik!

@krzysztofspilka – right now, the **Use filtering hooks** section mirrors the [Use sorting hooks](https://handsontable.com/docs/javascript-data-grid/rows-sorting/#use-sorting-hooks) section. But if we want to focus on the use cases mentioned, we can get rid of the **Use filtering hooks** section altogether, and add the following ones:
- **Block client-side filtering** (@AMBudnik mentioned it would be useful to have this info on the page)
- **Save filter settings** or **Retrieve filter conditions** (which, as discussed, was planned for [phase 2](https://github.com/handsontable/dev-handsontable/issues/1258))